pkg/tunnelengine/tunnelagent.go

```diff
@@ -203,6 +213,14 @@ func (c *TunnelHandler) shouldHandleGateway(gateway *v1beta1.Gateway) bool {
 		klog.InfoS("no public IP for gateway, waiting for sync", "gateway", klog.KObj(gateway))
 		return false
 	}
+	if getTunnelActiveEndpoints(gateway).NATType == "" {
+		klog.InfoS("no nat type for gateway, waiting for sync", "gateway", klog.KObj(gateway))
+		return false
+	}
+	if getTunnelActiveEndpoints(gateway).NATType != utils.NATSymmetric && getTunnelActiveEndpoints(gateway).PublicPort == 0 {
+		klog.InfoS("no public port for gateway, waiting for sync", "gateway", klog.KObj(gateway))
```

Comment:
Please reduce repeated calls to func

Reply:
Do you mean to call `getTunnelActiveEndpoints` only once in the `shouldHandleGateway` function to get the active endpoint?

Reply:
yes